### DISCUSSION TITLE
How to create AIR application deployment for MAC-OS

### DISCUSSION THREAD

**Discussion by aceinc2020** [airsdk/Adobe-Runtime-Support#3387] [Q&A]
I have a few applications written in FLEX converted from FLASH to AIR. During the conversion I created deployment packages for windows which includes the AIR Runtime and an SWF file which then loads and maintains the current versions of all of the remaining SWF files which comprise the application.

I have been tasked with providing an installer for MAC-OS. I have never worked with this OS. 

Where can I find help in creating a MAC-OS installer for my existing applications?

**Comment by ajwfrost** [comment]
There's an article here about it:
https://airsdk.dev/docs/tutorials/platform/macos/creating-macos-pkg-files

You'd need different certificates depending on whether you're going to be sending the PKG files to people directly, or putting this onto the App Store. (PKG is roughly similar to MSI). But otherwise it's a similar approach:
- create a "bundle" i.e. captive runtime application, using a macOS computer
- ensure everything is properly signed
- package up the resulting ".app" bundle (folder) using macOS tools into a PKG file
- notarize it so Apple can confirm it's all safe and signed

**Reply by aceinc2020** [reply]
Seems simple enough (heh heh). Having never touched a Mac, it may take a moment. 

I guess the first step is to buy one.

One area it looks like I will need to conquer before I get to the signing stage is;

> **It also assumes you already have a working publishing process for iOS and that you use IntelliJ on your Mac PC (although the steps can likely help anyone looking to take their air app to the Mac audience)**

My applications keeps them self up to date by examining each SWF file comparing a checksum and date against the version on the server, downloading new SWF files as needed before loading. Will this be an issue in  this OS?

**Reply by ajwfrost** [reply]
Yes, buying a mac might be a pre-requisite... we have a few Mac Minis for build machines, they're not bad and cheaper than the macbook option, but of course needing a separate screen and tied to a power socket..

For macOS, some of the complexity is around their code-signing and notarization process. And as part of that, they need to guarantee that your application doesn't change. So if you have code logic to check for updates like this, you will likely need to ensure that these updated SWF files are not stored in the signed application area (plus an end user may not have write permissions for that area anyway), so you can use the "app-storage" folder for this.

You should be able to do everything with normal macOS tools, terminal window (command prompt), finder (explorer), etc. But you'd also need an Apple Developer Program account in order to create "Developer ID" certificates that are used to sign your app, and the installer, to help validate the safety/accountability of the code.

If you run into anything slightly complex, please do ask, and maybe as part of this we can update that tutorial page to enhance it with other details!

thanks

**Reply by aceinc2020** [reply]
I have bought a Mac Mini. 

I am trying to create a .air file using Flash Builder 4.6, but I have a deprecation issue with getStyleDeclaration it tells me to use IStyleManager2.getStyleDeclaration but then I get an an undefined method error.
